internal/tasks/manager.go

```diff
@@ -0,0 +1,12 @@
+package tasks
+
+import "context"
+
+// IManager defines worker interface
+type IManager interface {
```

Comment:
the name sounds a little weird. just call this `Task` instead? we eventually only define `task.run()` with this.
Same with the file name as well?

Reply:
Yes, Needed suggestions here

Reply:
Can we use Task then? As discussed in the future, multiple `Tasks` can be tied together onto one `Workflow` and run by a `Manager`.

Reply:
Yes, Sounds good, but want to keep that in separate PR

Reply:
Sure. Rename at least in this PR? `IManager` really sounds out of place.

Reply:
Updated